### PR TITLE
Check basic driver function after main_test executed

### PR DESCRIPTION
**Pull request by lijinlijin**
1. Check basic driver function after main_test executed 
2. Adjust viorng config to optimize rng cases.

ID: 1725565

Signed-off-by: Li Jin <lijin@redhat.com>

### CODE REVIEW

**Comment by lijinlijin** [comment]
(1/2) Host_RHEL.m8.u1.virtio_rng.rng_egd.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.viorng_in_use.with_live_migration.after_bg_test: PASS (193.31 s)
 (2/2) Host_RHEL.m8.u1.virtio_rng.rng_egd.qcow2.virtio_scsi.up.virtio_net.Guest.Win2008.x86_64.r2.io-github-autotest-qemu.viorng_in_use.with_live_migration.after_bg_test: PASS (160.12 s)

**Comment by lijinlijin** [comment]
@ybduan 
Could you help to review it?
Thanks

**Comment by ybduan** [comment]
I think we need all results of "--testcase=viorng_in_use --rng_type=virtio_rng".

**Comment by lijinlijin** [comment]
All cases passed, following is the result
'01-Host_RHEL.m8.u1.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.viorng_in_use.with_stop_continue.before_bg_test' Finshed 2019-07-01 05:54:37 2019-07-01 05:55:58 PASS 80.98191
'02-Host_RHEL.m8.u1.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.viorng_in_use.with_stop_continue.during_bg_test' Finshed 2019-07-01 05:55:58 2019-07-01 05:58:27 PASS 149.429962
'03-Host_RHEL.m8.u1.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.viorng_in_use.with_stop_continue.after_bg_test' Finshed 2019-07-01 05:58:27 2019-07-01 05:59:41 PASS 73.885768
'04-Host_RHEL.m8.u1.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.viorng_in_use.with_shutdown.before_bg_test' Finshed 2019-07-01 05:59:41 2019-07-01 06:01:37 PASS 115.909834
'05-Host_RHEL.m8.u1.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.viorng_in_use.with_shutdown.during_bg_test' Finshed 2019-07-01 06:01:37 2019-07-01 06:02:38 PASS 60.696297
'06-Host_RHEL.m8.u1.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.viorng_in_use.with_shutdown.after_bg_test' Finshed 2019-07-01 06:02:38 2019-07-01 06:03:42 PASS 64.23417
'07-Host_RHEL.m8.u1.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.viorng_in_use.with_reboot.before_bg_test' Finshed 2019-07-01 06:03:42 2019-07-01 06:06:08 PASS 145.709501
'08-Host_RHEL.m8.u1.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.viorng_in_use.with_reboot.during_bg_test' Finshed 2019-07-01 06:06:08 2019-07-01 06:09:03 PASS 175.182209
'09-Host_RHEL.m8.u1.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.viorng_in_use.with_reboot.after_bg_test' Finshed 2019-07-01 06:09:03 2019-07-01 06:11:19 PASS 136.021832
'10-Host_RHEL.m8.u1.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.viorng_in_use.with_system_reset.before_bg_test' Finshed 2019-07-01 06:11:19 2019-07-01 06:13:38 PASS 138.677186
'11-Host_RHEL.m8.u1.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.viorng_in_use.with_system_reset.during_bg_test' Finshed 2019-07-01 06:13:38 2019-07-01 06:16:42 PASS 184.023689
'12-Host_RHEL.m8.u1.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.viorng_in_use.with_system_reset.after_bg_test' Finshed 2019-07-01 06:16:42 2019-07-01 06:18:50 PASS 127.892109
'13-Host_RHEL.m8.u1.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.viorng_in_use.with_live_migration.before_bg_test' Finshed 2019-07-01 06:18:50 2019-07-01 06:21:49 PASS 179.347387
'14-Host_RHEL.m8.u1.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.viorng_in_use.with_live_migration.during_bg_test' Finshed 2019-07-01 06:21:49 2019-07-01 06:25:38 PASS 229.081004
'15-Host_RHEL.m8.u1.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.viorng_in_use.with_live_migration.after_bg_test' Finshed 2019-07-01 06:25:38 2019-07-01 06:28:47 PASS 188.451791
'16-Host_RHEL.m8.u1.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.Win2008.x86_64.r2.io-github-autotest-qemu.viorng_in_use.with_stop_continue.before_bg_test' Finshed 2019-07-01 06:28:47 2019-07-01 06:35:23 PASS 396.461808
'17-Host_RHEL.m8.u1.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.Win2008.x86_64.r2.io-github-autotest-qemu.viorng_in_use.with_stop_continue.during_bg_test' Finshed 2019-07-01 06:35:23 2019-07-01 06:42:35 PASS (126.73 s)
'18-Host_RHEL.m8.u1.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.Win2008.x86_64.r2.io-github-autotest-qemu.viorng_in_use.with_stop_continue.after_bg_test' Finshed 2019-07-01 06:42:35 2019-07-01 06:43:44 PASS 68.846366
'19-Host_RHEL.m8.u1.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.Win2008.x86_64.r2.io-github-autotest-qemu.viorng_in_use.with_shutdown.before_bg_test' Finshed 2019-07-01 06:43:44 2019-07-01 06:45:45 PASS 121.206644
'20-Host_RHEL.m8.u1.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.Win2008.x86_64.r2.io-github-autotest-qemu.viorng_in_use.with_shutdown.during_bg_test' Finshed 2019-07-01 06:45:45 2019-07-01 06:46:56 PASS 71.474323
'21-Host_RHEL.m8.u1.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.Win2008.x86_64.r2.io-github-autotest-qemu.viorng_in_use.with_shutdown.after_bg_test' Finshed 2019-07-01 06:46:56 2019-07-01 06:47:56 PASS 59.253013
'22-Host_RHEL.m8.u1.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.Win2008.x86_64.r2.io-github-autotest-qemu.viorng_in_use.with_reboot.before_bg_test' Finshed 2019-07-01 06:47:56 2019-07-01 06:49:51 PASS 115.279764
'23-Host_RHEL.m8.u1.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.Win2008.x86_64.r2.io-github-autotest-qemu.viorng_in_use.with_reboot.during_bg_test' Finshed 2019-07-01 06:49:51 2019-07-01 06:52:04 PASS 133.448762
'24-Host_RHEL.m8.u1.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.Win2008.x86_64.r2.io-github-autotest-qemu.viorng_in_use.with_reboot.after_bg_test' Finshed 2019-07-01 06:52:04 2019-07-01 06:53:53 PASS 108.912733
'25-Host_RHEL.m8.u1.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.Win2008.x86_64.r2.io-github-autotest-qemu.viorng_in_use.with_system_reset.before_bg_test' Finshed 2019-07-01 06:53:53 2019-07-01 06:55:53 PASS 119.393284
'26-Host_RHEL.m8.u1.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.Win2008.x86_64.r2.io-github-autotest-qemu.viorng_in_use.with_system_reset.during_bg_test' Finshed 2019-07-01 06:55:53 2019-07-01 07:03:25 PASS 451.864522
'27-Host_RHEL.m8.u1.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.Win2008.x86_64.r2.io-github-autotest-qemu.viorng_in_use.with_system_reset.after_bg_test' Finshed 2019-07-01 07:03:25 2019-07-01 07:05:27 PASS 122.425393
'28-Host_RHEL.m8.u1.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.Win2008.x86_64.r2.io-github-autotest-qemu.viorng_in_use.with_live_migration.before_bg_test' Finshed 2019-07-01 07:05:27 2019-07-01 07:12:33 PASS (119.11 s)
'29-Host_RHEL.m8.u1.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.Win2008.x86_64.r2.io-github-autotest-qemu.viorng_in_use.with_live_migration.during_bg_test' Finshed 2019-07-01 07:12:33 2019-07-01 07:15:46 PASS 193.40547   
'30-Host_RHEL.m8.u1.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.Win2008.x86_64.r2.io-github-autotest-qemu.viorng_in_use.with_live_migration.after_bg_test' Finshed 2019-07-01 07:15:46 2019-07-01 07:17:38 PASS 112.335704   
'31-Host_RHEL.m8.u1.virtio_rng.rng_egd.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.viorng_in_use.with_stop_continue.after_bg_test' Finshed 2019-07-01 07:17:38 2019-07-01 07:24:43 PASS 424.296915         
'32-Host_RHEL.m8.u1.virtio_rng.rng_egd.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.viorng_in_use.with_shutdown.after_bg_test' Finshed 2019-07-01 07:24:43 2019-07-01 07:25:55 PASS 72.717969               
'33-Host_RHEL.m8.u1.virtio_rng.rng_egd.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.viorng_in_use.with_reboot.after_bg_test' Finshed 2019-07-01 07:25:55 2019-07-01 07:28:29 PASS 153.114732                
'34-Host_RHEL.m8.u1.virtio_rng.rng_egd.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.viorng_in_use.with_system_reset.after_bg_test' Finshed 2019-07-01 07:28:29 2019-07-01 07:30:54 PASS 145.118138          
'35-Host_RHEL.m8.u1.virtio_rng.rng_egd.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.viorng_in_use.with_live_migration.after_bg_test' Finshed 2019-07-01 07:30:54 2019-07-01 07:34:09 PASS 195.03824         
'36-Host_RHEL.m8.u1.virtio_rng.rng_egd.qcow2.virtio_scsi.up.virtio_net.Guest.Win2008.x86_64.r2.io-github-autotest-qemu.viorng_in_use.with_stop_continue.after_bg_test' Finshed 2019-07-01 07:34:09 2019-07-01 07:36:10 PASS 121.398125       
'37-Host_RHEL.m8.u1.virtio_rng.rng_egd.qcow2.virtio_scsi.up.virtio_net.Guest.Win2008.x86_64.r2.io-github-autotest-qemu.viorng_in_use.with_shutdown.after_bg_test' Finshed 2019-07-01 07:36:10 2019-07-01 07:37:29 PASS 78.766726             
'38-Host_RHEL.m8.u1.virtio_rng.rng_egd.qcow2.virtio_scsi.up.virtio_net.Guest.Win2008.x86_64.r2.io-github-autotest-qemu.viorng_in_use.with_reboot.after_bg_test' Finshed 2019-07-01 07:37:29 2019-07-01 07:39:27 PASS 118.313267              
'39-Host_RHEL.m8.u1.virtio_rng.rng_egd.qcow2.virtio_scsi.up.virtio_net.Guest.Win2008.x86_64.r2.io-github-autotest-qemu.viorng_in_use.with_system_reset.after_bg_test' Finshed 2019-07-01 07:39:27 2019-07-01 07:41:36 PASS 128.794513        
'40-Host_RHEL.m8.u1.virtio_rng.rng_egd.qcow2.virtio_scsi.up.virtio_net.Guest.Win2008.x86_64.r2.io-github-autotest-qemu.viorng_in_use.with_live_migration.after_bg_test' Finshed 2019-07-01 07:41:36 2019-07-01 07:43:41 PASS 124.528487

**Comment by ybduan** [comment]
LGTM.
Acked-by: Yanbin Duan yduan@redhat.com

**Comment by lijinlijin** [comment]
@yanan-fu 
Could you help to have a review?
Thanks

**Comment by yanan-fu** [comment]
@lijinlijin  I do not see any new added cases but remove `rng_egd`  for case `before_bg_test` and `during_bg_test`,  but  this patch is for `migrate guest when rng-egd backend is used`.
could you explain more, thanks !

File: qemu/tests/cfg/viorng_in_use.cfg [hunk 1]
```diff
@@ -23,7 +23,7 @@
             driver_id_cmd = 'WIN_UTILS:\devcon\wnet_amd64\devcon.exe find * | find "VirtIO"'
             driver_check_cmd = "WIN_UTILS:\devcon\wnet_amd64\devcon.exe status @DRIVER_ID"
     Linux:
-        read_rng_cmd  = "dd if=/dev/random bs=10 count=1000000 2>/dev/null|hexdump"
+        read_rng_cmd  = "dd if=/dev/random bs=10 count=10 2>/dev/null|hexdump"
```
Comment (yanan-fu) [inline qemu/tests/cfg/viorng_in_use.cfg]: Why change the count ?

File: qemu/tests/cfg/viorng_in_use.cfg [hunk 2]
```diff
@@ -14,7 +14,7 @@
         target_process = random\w*.exe
         rng_data_rex = "0x\w"
         driver_name = "viorng"
-        read_rng_cmd = for /l %i in (1, 1, 1000) do "WIN_UTILS:\random_%PROCESSOR_ARCHITECTURE%.exe"
```
Comment (yanan-fu) [inline qemu/tests/cfg/viorng_in_use.cfg]: why only run once instead of 1000 times ?

Comment (lijinlijin) [reply qemu/tests/cfg/viorng_in_use.cfg]: The aim of the command is just to verify rng basic function works well.
No need to run so many times, it takes longer time and egd performance is not good, run 1000 times will make this cmd always timeout.

File: qemu/tests/cfg/viorng_in_use.cfg (hunk 1 above)
Comment (lijinlijin) [reply qemu/tests/cfg/viorng_in_use.cfg]: The same reason as above.

File: qemu/tests/cfg/viorng_in_use.cfg (hunk 2 above)
Comment (yanan-fu) [reply qemu/tests/cfg/viorng_in_use.cfg]: make sense

**Comment by lijinlijin** [comment]
@yanan-fu 
The main part of adding a new case is adding run_bg_test_sep(bg_stress_test) after run_virt_sub_test().

Adding "no rng_egd" in  before_bg_test and  during_bg_test variants is because we do not have such cases in egd backend

**Comment by lijinlijin** [comment]
@yanan-fu 
As you suggested, separate the change into two commits, thanks a lot.

**Comment by yanan-fu** [comment]
@lijinlijin  Not only migration, other tests such as `with_stop_continue`, `with_shtdown` ...all need to run stress test again after main test finished ?

**Comment by lijinlijin** [comment]
@yanan-fu 
Yes, function test should be checked after each subtest to make sure it works well.
shutdown case will not run the test as vm is not alive.

**Comment by yanan-fu** [comment]
@lijinlijin  checked the polarion cases, it is about `during`. In automation, `during_bg_test` indeed  run stress test after main test finished.   Do we still need this change for `after_bg_test` ?

**Comment by yanan-fu** [comment]
LGTM
Acked-by: Yanan Fu <yfu@redhat.com>

**Pull request merged**